cypress-io/cypress
# chore: trigger run notifications

- Closes https://github.com/cypress-io/cypress/issues/26788

### Additional details


This PR adds a new class of actions that allows us to trigger different kinds of notifications. These actions are used in `RelevantRunsDataSource` to send notifications to the user when it is appropriate based on the requirements in the ClickUp. See the related issue for links to the ClickUp and Figma.

I'm using the `chore` prefix in the PR title just so that the changelog step doesn't yell at me. These messages will all be squashed into one commit when we merge the feature branch into develop.

### Steps to test


1. Run Cypress with a project that is connected to Cloud and has at least one run recorded
2. Click "Enable desktop notifications" in the banner at the top of the page (if you've dismissed this banner before you'll need to set `desktopNotificationsEnabled: null` in your global `state.json` file)
3. Break a test in the project (so that the run will fail)
4. Kick off a new test run for the project and leave the Cypress app running
5. Verify that after the first test fails that you see a push notification saying that your run started failing
6. Verify that clicking the notification takes you to the debug page and focuses the run in question
7. Verify that after the run completes you see another push notification saying that your run has failed

### How has the user experience changed?


### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)?
